cmd/k8s-operator/ingress-for-pg.go

```diff
@@ -411,8 +417,15 @@ func (r *HAIngressReconciler) maybeCleanupProxyGroup(ctx context.Context, proxyG
 			if err = r.maybeUpdateAdvertiseServicesConfig(ctx, proxyGroupName, vipServiceName, false, logger); err != nil {
 				return false, fmt.Errorf("failed to update tailscaled config services: %w", err)
 			}
-			delete(cfg.Services, vipServiceName)
-			serveConfigChanged = true
+			svcCfg, ok := cfg.Services[vipServiceName]
+			if ok {
+				logger.Infof("Removing VIPService %q from serve config", vipServiceName)
+				delete(cfg.Services, vipServiceName)
+				serveConfigChanged = true
+			}
+			if err := r.cleanupCertResources(ctx, proxyGroupName, svcCfg); err != nil {
```

Comment:
I think this should maybe move to being the first statement inside 'if ok' above. If we partially fail during this cleanup, we won't retry on the next reconcile because the serve config we rely on already got updated.

Reply:
This https://github.com/tailscale/tailscale/pull/15308#discussion_r2002561656 should fix it